gax/src/streamingRetryRequest.ts

```diff
@@ -41,15 +41,6 @@ interface streamingRetryRequestOptions {
 export function streamingRetryRequest(opts: streamingRetryRequestOptions) {
   opts = Object.assign({}, DEFAULTS, opts);
 
-  if (opts.request === undefined) {
```

Comment:
It seems like we're just assuming it will be provided. Given that we have testing for this, perhaps it's safer to at least throw the error if `opts.request === undefined`, and just remove requiring `request`